### DISCUSSION TITLE
Connect to my custom chain with chain_id: 1

### DISCUSSION THREAD

**Discussion by PoratRoy** [rainbow-me/rainbowkit#1309] [General]
Enable to connect my custom chain network with the default chain `1`

When I'm trying to switch chains for my custom chain with chain_id 1, instead of the `Allow this site to add a network?` alert, I'm connecting straight to the Ethereum mainnet network.

```
const customChain: Chain = {
  id: 0x1,
  name: "My Custom Chain",
  network: "custom-chain",
  nativeCurrency: { decimals: 18, name: 'ETH', symbol: 'ETH' },
  rpcUrls: {
    default: { http: ["https://xxx.xxx"] },
    public: { http: ["https://xxx.xxx"] }
  },
  testnet: false
}
```

```
const { chains, publicClient } = configureChains(
	[ customChain ],
	[ jsonRpcProvider ( { rpc: (chain) => ( { http: chain.rpcUrls.default.http[0] } ) } ) ]
);
```

I'm using 
react: 18.2.0
wagmi: 1.2.0
rainbow-me/rainbowkit: 1.0.2

when I'm trying to connect manually, the connection works.
What am I missing?

**Comment by glitch-txs** [comment]
Have you tried using another id? Mainnet is usually hardcoded on Ethereum wallets

**Reply by PoratRoy** [reply]
Hey @glitch-txs, yes, and it works. the only problem is that I have to use id`1` for my project.
maybe there is a way to force the chain to use 1?

**Reply by glitch-txs** [reply]
It think this depends on how the wallet works internally
